protocol/src/main/kotlin/models/Resource.kt

```diff
@@ -108,10 +113,13 @@ sealed class Resource {
 
       // TODO json schema validation using specific type schema
 
-      return when (kindEnum) {
-        ResourceKind.offering -> objectMapper.readValue<Offering>(payload)
-        ResourceKind.reputation -> TODO()
+      val resource = when (kindEnum) {
+        ResourceKind.offering -> jsonMapper.readValue<Offering>(payload)
+        ResourceKind.reputation -> throw NotImplementedError()
```

Comment:
just fyi, in the validate json schema branch, i removed reputation as a resource kind, because we loop through the resourcekind and try to map it to `reputation.schema.json` which does not exist. 

but, i guess we could make an empty one in tbdex and pull it over to this repo.

Reply:
see https://github.com/TBD54566975/tbdex/pull/166